chainer/iterators/multiprocess_iterator.py

```diff
@@ -48,13 +50,21 @@ class MultiprocessIterator(iterator.Iterator):
         n_prefetch (int): Number of prefetch batches.
         shared_mem (int): The size of using shared memory per data.
             If ``None``, size is adjusted automatically.
+        order_sampler (callable): A callable that generates the order
+            of the indices to sample in the next epoch when a epoch finishes.
+            This function should take two arguements: the current order
+            and the current position of the iterator.
+            This should return the next order, which should have the
+            same length as the dataset.
```

Comment:
Do you think we impose that the returned value should not have duplicated values?

Reply:
I do not think it is necessary to prohibit duplicated values.
First of all, the rest of the code works without this constraint.
Second, it becomes necessary to support duplicate values in some application. For instance, in a situation where a user wants to sample some indices in a biased way.